js/widgets/forms/textinput.js

```diff
@@ -82,56 +89,30 @@ $.widget( "mobile.textinput", {
 	},
 
 	_enhance: function() {
-		var elementClasses = [];
+		var outer;
 
-		if ( this.isTextarea ) {
-			elementClasses.push( "ui-input-text" );
-		}
-
-		if ( this.isTextarea || this.isRange ) {
-			elementClasses.push( "ui-shadow-inset" );
-		}
-
-		//"search" and "text" input widgets
 		if ( this.inputNeedsWrap ) {
-			this.element.wrap( this._wrap() );
+			outer = $( "<div>" );
+			if ( this.isSearch ) {
+				this._searchIcon = $( "<span>" ).appendTo( outer );
+				this._addClass( this._searchIcon, "ui-textinput-search-icon" );
```

Comment:
The span is appended to the wrapper div, but the wrapper div is not appended to the DOM, so this won't cause a reflow.